crates/bevy_pbr/src/render/light.rs

```diff
@@ -770,7 +797,9 @@ pub fn prepare_lights(
             color: Vec4::from_slice(&light.color.to_f32_array()) * light.illuminance,
             // direction is negated to be ready for N.L
             dir_to_light: light.transform.back().into(),
-            flags: flags.bits(),
+            flags: flags.bits()
+                | (light.render_layers.bits_compact_lossy() as u32)
```

Comment:
I know the code is pretty self evidence but it might be nice to leave a breadcrumb here for future readers (e.g., something like "these aren't flags, but are using the extra space").